File: C/Iniciante/1045tiposdetriangulos.c
#include <stdio.h>
#include <math.h>
int main() {
    double a,b,c;
    scanf("%lf%lf%lf",&a,&b,&c);
    if (a < c) {
        double tmp = c;
        c = a;
        a = tmp;
    }
    if (a < b) {
        double tmp = b;
        b = a;
        a = tmp;
    }
    if (b < c) {
        double tmp = c;
        c = b;
        b = tmp;
    }
    if(a>=b+c){
        printf("NAO FORMA TRIANGULO\n");
    }else {
        if (pow(a, 2) == pow(b, 2) + pow(c, 2)) {
            printf("TRIANGULO RETANGULO\n");
        }
        if (pow(a, 2) > pow(b, 2) + pow(c, 2)) {
            printf("TRIANGULO OBTUSANGULO\n");
        }
        if (pow(a, 2) < pow(b, 2) + pow(c, 2)) {
            printf("TRIANGULO ACUTANGULO\n");
        }
        if (a == b && b == c) {
            printf("TRIANGULO EQUILATERO\n");
        }
        if ((a == b && a != c) || (a == c && a != b) || (b==c && b!=a)) {
            printf("TRIANGULO ISOSCELES\n");
        }
    }
    return 0;
}

File: Python/iniciante/1045tiposdetriangulos.py
import math as m
linha = input().split(" ")
a, b, c = linha
a = float(a)
b = float(b)
c = float(c)
if a < c:
    tmp = c
    c = a
    a = tmp
if a < b:
    tmp = b
    b = a
    a = tmp
if b < c:
    tmp = c
    c = b
    b = tmp
if a >= b+c:
    print("NAO FORMA TRIANGULO")
else:
    if m.pow(a, 2) == m.pow(b, 2) + m.pow(c, 2):
        print("TRIANGULO RETANGULO")
    if m.pow(a, 2) > m.pow(b, 2) + m.pow(c, 2):
        print("TRIANGULO OBTUSANGULO")
    if m.pow(a, 2) < m.pow(b, 2) + m.pow(c, 2):
        print("TRIANGULO ACUTANGULO")
    if a == b and b == c:
        print("TRIANGULO EQUILATERO")
    if (a == b and a != c) or (a == c and a != b) or (b == c and b != a):
        print("TRIANGULO ISOSCELES")
